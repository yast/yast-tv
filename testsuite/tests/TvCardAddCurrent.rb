# encoding: utf-8

module Yast
  class TvCardAddCurrentClient < Client
    def main
      # testedfiles: Tv.ycp
      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      # Tv::current_card == nil
      TEST_It_Now(nil)
      # Tv::current_card == $[]
      TEST_It_Now({})

      # Tv::current_card == something useful
      TEST_It_Now(
        {
          "module"        => "abc",
          "parameters"    => { "par_par" => "val_val" },
          "unique_key"    => "abc.uniq_key",
          "sound_card_no" => 5
        }
      )

      nil
    end

    def TEST_It_Now(curcard_totest)
      curcard_totest = deep_copy(curcard_totest)
      DUMP("-------------------------------------------")
      Tv.cards = nil
      Tv.current_card = deep_copy(curcard_totest)
      Tv.CardAddCurrent
      DUMP(Tv.cards)

      # Tv::cards == $[]
      Tv.cards = []
      Tv.current_card = deep_copy(curcard_totest)
      Tv.CardAddCurrent
      DUMP(Tv.cards)

      # Tv::cards == card1
      Tv.cards = [
        {
          "module"        => "bttv",
          "parameters"    => { "param1" => "testvalue" },
          "unique_key"    => "hluk.HvsIesmXXX1",
          "sound_card_no" => 0
        }
      ]
      Tv.current_card = deep_copy(curcard_totest)
      Tv.CardAddCurrent
      DUMP(Tv.cards)

      # Tv::cards == card1, card2
      Tv.cards = [
        {
          "module"        => "bttv",
          "parameters"    => { "param1" => "testvalue" },
          "unique_key"    => "hluk.HvsIesmXXX1",
          "sound_card_no" => 0
        },
        {
          "module"        => "bttv",
          "parameters"    => { "par_other" => "testvalue2" },
          "unique_key"    => "j7F3.J8dhgs34fK2",
          "sound_card_no" => 1
        }
      ]
      Tv.current_card = deep_copy(curcard_totest)
      Tv.CardAddCurrent
      DUMP(Tv.cards)

      # Tv::cards == card1, nil, nil, card4
      Tv.cards = [
        {
          "module"        => "bttv",
          "parameters"    => { "param1" => "testvalue" },
          "unique_key"    => "hluk.HvsIesmXXX1",
          "sound_card_no" => 0
        },
        {},
        {},
        {
          "module"        => "xyz",
          "parameters"    => { "par_other" => "testvalue2" },
          "unique_key"    => "j7F3.J8dhgs34fK2",
          "sound_card_no" => 1
        }
      ]
      Tv.current_card = deep_copy(curcard_totest)
      Tv.CardAddCurrent
      DUMP(Tv.cards)

      nil
    end
  end
end

Yast::TvCardAddCurrentClient.new.main
