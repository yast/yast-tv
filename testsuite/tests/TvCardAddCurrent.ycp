{

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size" : -1 ],
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

    define TEST_It_Now(map curcard_totest) ``{
	// Tv::cards == nil
	Tv::cards = nil;
	Tv::current_card = curcard_totest;
	Tv::CardAddCurrent();
	DUMP(Tv::cards);

	// Tv::cards == $[]
	Tv::cards = [];
	Tv::current_card = curcard_totest;
	Tv::CardAddCurrent();
	DUMP(Tv::cards);

	// Tv::cards == card1
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ]
	    ];
	Tv::current_card = curcard_totest;
	Tv::CardAddCurrent();
	DUMP(Tv::cards);

	// Tv::cards == card1, card2
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ],
	      $[ "module" : "bttv",
		 "parameters" : $[ "par_other" : "testvalue2" ],
		 "unique_key" : "j7F3.J8dhgs34fK2",
		 "sound_card_no" : 1
	      ]
	    ];
	Tv::current_card = curcard_totest;
	Tv::CardAddCurrent();
	DUMP(Tv::cards);

	// Tv::cards == card1, nil, nil, card4
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ],
	      nil,
	      nil,
	      $[ "module" : "xyz",
		 "parameters" : $[ "par_other" : "testvalue2" ],
		 "unique_key" : "j7F3.J8dhgs34fK2",
		 "sound_card_no" : 1
	      ]
	    ];
	Tv::current_card = curcard_totest;
	Tv::CardAddCurrent();
	DUMP(Tv::cards);
    }

    // Tv::current_card == nil
    TEST_It_Now(nil);

    // Tv::current_card == $[]
    TEST_It_Now($[]);

    // Tv::current_card == something useful
    TEST_It_Now($[ "module" : "abc",
		   "parameters" : $[ "par_par" : "val_val" ],
		   "unique_key" : "abc.uniq_key",
		   "sound_card_no" : 5
		]);
}
