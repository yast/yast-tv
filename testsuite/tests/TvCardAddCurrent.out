Read	.target.ycp ["/var/lib/YaST2/installmap.ycp", $[]] $[]
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Dump	[nil]
Dump	[nil]
Dump	[$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"], nil]
Dump	[$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"], $["module":"bttv", "parameters":$["par_other":"testvalue2"], "sound_card_no":1, "unique_key":"j7F3.J8dhgs34fK2"], nil]
Dump	[$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"], nil, nil, $["module":"xyz", "parameters":$["par_other":"testvalue2"], "sound_card_no":1, "unique_key":"j7F3.J8dhgs34fK2"]]
Dump	[$[]]
Dump	[$[]]
Dump	[$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"], $[]]
Dump	[$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"], $["module":"bttv", "parameters":$["par_other":"testvalue2"], "sound_card_no":1, "unique_key":"j7F3.J8dhgs34fK2"], $[]]
Dump	[$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"], $[], nil, $["module":"xyz", "parameters":$["par_other":"testvalue2"], "sound_card_no":1, "unique_key":"j7F3.J8dhgs34fK2"]]
Dump	[$["module":"abc", "parameters":$["par_par":"val_val"], "sound_card_no":5, "unique_key":"abc.uniq_key"]]
Dump	[$["module":"abc", "parameters":$["par_par":"val_val"], "sound_card_no":5, "unique_key":"abc.uniq_key"]]
Dump	[$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"], $["module":"abc", "parameters":$["par_par":"val_val"], "sound_card_no":5, "unique_key":"abc.uniq_key"]]
Dump	[$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"], $["module":"bttv", "parameters":$["par_other":"testvalue2"], "sound_card_no":1, "unique_key":"j7F3.J8dhgs34fK2"], $["module":"abc", "parameters":$["par_par":"val_val"], "sound_card_no":5, "unique_key":"abc.uniq_key"]]
Dump	[$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"], $["module":"abc", "parameters":$["par_par":"val_val"], "sound_card_no":5, "unique_key":"abc.uniq_key"], nil, $["module":"xyz", "parameters":$["par_other":"testvalue2"], "sound_card_no":1, "unique_key":"j7F3.J8dhgs34fK2"]]
